src/wifi.cpp

```diff
@@ -0,0 +1,694 @@
+
+// Define the number of currently supported TCP commands
+#define NUM_TCP_COMMANDS 3
+
+// Define, when receiving a TCP packet, what special strings should cause what actions
+typedef struct tcp_command_s {
+    // The string, that if the TCP packet matches, should trigger an action
+    const char* command;
+    // The action to be triggered if the TCP packet matched command
+    void (*action)();
+} tcp_command_t;
+
+// ======================= //
+// Instantiate useful data //
+// ======================= //
+
+// Keep track of how many times the device has retried connecting to one AP in a row
+size_t num_wifi_connect_retries = 0;
+// Keep track of the handle to the WiFi event group, so we can set success and failure bits on its status
+EventGroupHandle_t event_group_handle_wifi = nullptr;
+// Keep track of the file handle being used as the IP socket connection
+// TODO: Put a mutex around this file handle
+int ip_socket_file_descriptor = 0;
+// Keep track of the handle of the task that reads IP packets (task_read_ip_packets(...))
+TaskHandle_t read_ip_packet_task_handle = nullptr;
+// Keep track of the esp_netif object attaching netif to wifi and registering wifi handlers to the default event loop
+void *esp_netif = nullptr;
+
+// Keep track of the special strings that if a TCP packet matches,
+// should excute an action, and what action they should ewxecute
+tcp_command_t tcp_commands[NUM_TCP_COMMANDS] = {
+    {
+        .command = "up",
+        .action = []() { add_to_menu_input_queue(
+            /* MENU_INPUT_t menu_input = */ MENU_INPUT_UP,
+            /* bool from_isr = */ false); },
+    },
+    {
+        .command = "down",
+        .action = []() { add_to_menu_input_queue(
+            /* MENU_INPUT_t menu_input = */ MENU_INPUT_DOWN,
+            /* bool from_isr = */ false); },
+    },
+    {
+        .command = "confirm",
+        .action = []() { add_to_menu_input_queue(
+            /* MENU_INPUT_t menu_input = */ MENU_INPUT_CONFIRM,
+            /* bool from_isr = */ false); },
+    },
+#if 0
+    {
+        .command = "sleep",
+        .action = []() { add_to_menu_input_queue(
+            /* MENU_INPUT_t menu_input = */ MENU_INPUT_SLEEP,
+            /* bool from_isr = */ false); },
+    },
+#endif
+};
+
+// ============ //
+// Define tasks //
+// ============ //
+
+// A task whose job it is to read all incoming TCP packets over the connected IP socket.
+// Based on the contents of the TCP packet, it may execute certain actions.
+// Once the socket the TCP connection is set up on, ip_socket_file_descriptor,
+// is no longer readable, it automatically kills itself and resets the handle.
+void task_read_ip_packets()
+{
+    // Create a 0-initialized buffer IP packets will be read into
+    char read_buffer[16] = { 0 };
+    int num_read_bytes = 0;
+
+    while(1)
+    {
+        // Get the number of bytes in the file descriptor,
+        // copy up to count bytes from the file descriptor to our read buffer
```

Comment:
count bytes? clarify, num_read_bytes?

Reply:
addressed in: https://github.com/jCallon/squirt/pull/1/commits/edb88a636d2cf54a42f970d190cd054c04993412